pkg/asset/manifests/dns.go

```diff
@@ -134,14 +134,37 @@ func (d *DNS) Generate(dependencies asset.Parents) error {
 			}
 		}
 	case gcptypes.Name:
+		// Set the public zone
 		if installConfig.Config.Publish == types.ExternalPublishingStrategy {
-			zone, err := icgcp.GetPublicZone(context.TODO(), installConfig.Config.Platform.GCP.ProjectID, installConfig.Config.BaseDomain)
+
+			if installConfig.Config.GCP.PublicManagedZone == "" {
+				zone, err := icgcp.GetPublicZone(context.TODO(), installConfig.Config.Platform.GCP.ProjectID, installConfig.Config.BaseDomain)
+				if err != nil {
+					return errors.Wrapf(err, "failed to get public zone for %q", installConfig.Config.BaseDomain)
+				}
+				config.Spec.PublicZone = &configv1.DNSZone{ID: zone.Name}
+			} else if !strings.EqualFold(installConfig.Config.GCP.PublicManagedZone, "None") {
+				// When "None" manifest data is NOT set
+				zone, err := icgcp.GetZoneByName(context.TODO(), installConfig.Config.Platform.GCP.ProjectID, installConfig.Config.GCP.PublicManagedZone)
+				if err != nil {
+					return errors.Wrapf(err, "failed to get public zone %q", installConfig.Config.GCP.PublicManagedZone)
+				}
+				config.Spec.PublicZone = &configv1.DNSZone{ID: zone.Name}
+			}
+		}
+
+		// Set the private zone
+		if installConfig.Config.GCP.PrivateManagedZone == "" {
+			config.Spec.PrivateZone = &configv1.DNSZone{ID: fmt.Sprintf("%s-private-zone", clusterID.InfraID)}
+		} else if !strings.EqualFold(installConfig.Config.GCP.PrivateManagedZone, "None") {
+			// When "None" manifest data is NOT set
+			_, zoneName, err := icgcp.ParseZone(installConfig.Config.Platform.GCP.ProjectID, installConfig.Config.GCP.PrivateManagedZone)
```

Comment:
No need to parse here, let's pass the `installConfig.Config.GCP.PrivateManagedZone` into the manifest
```
config.Spec.PublicZone = &configv1.DNSZone{ID: installConfig.Config.GCP.PrivateManagedZone}
```